web-client/src/client/lib.rs

```diff
@@ -569,15 +595,7 @@ impl Client {
             None
         };
 
-        // Unsubscribe from any address we subscribed to, without caring about the result
-        if let Some(address) = subscribed_address {
-            let owned_consensus = consensus.clone();
-            spawn_local(async move {
-                let _ = owned_consensus
-                    .unsubscribe_from_addresses(vec![address], 1)
-                    .await;
-            });
-        }
+        mem::drop(address_subscription);
```

Comment:
This will be implicitly executed as soon as the scope ends. Is there reason to do it earlier than that?

Reply:
Otherwise Clippy complains the variable is never used and suggests to prefix it with an underscore. But then I fear it is compiled out?

Reply:
The call to the drop implementation should make it not unused in the resulting representation. Thus even if underscored it should not get optimized away as it has observable behaviour attached to it (the drop impl). 

I am not entirely sure however if that is in fact true, so maybe keep it like it is.

Reply:
> The call to the drop implementation should make it not unused in the resulting representation. Thus even if underscored it should not get optimized away as it has observable behaviour attached to it (the drop impl).
> 
> I am not entirely sure however if that is in fact true, so maybe keep it like it is.

Yes, if underscored, it doesn't get optimized away. Note that this is different from the name which only consists of an underscore `_` which means "immediately drop this".

I think the `mem::drop` is still fine, this way this PR doesn't change when the address is unsubscribed from in the happy case.

Reply:
Yeah I don't mind it either. More of a curiosity ask.